src/core/vfm/metastore_test.go

```diff
@@ -0,0 +1,480 @@
+package vfm
+
+import (
+	"context"
+	"errors"
+	"testing"
+)
+
+func FatalIfErr(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+var ErrIncorrectError = errors.New("Expected error was not thrown!")
+
+func ExpectErr(t *testing.T, actual error, expected error) {
+	if actual != expected {
+		t.Logf("Actual Error : %s\n", actual.Error())
+		t.Fatal(ErrIncorrectError)
+	}
+}
+
+var ErrConditionNotMet = errors.New("Condition not met!")
+
+func ExpectTrue(t *testing.T, res bool) {
+	if !res {
+		t.Log("Condition Not Met!")
+		t.Fatal(ErrConditionNotMet)
+	}
+}
+
+// We have this test all function to ensure clean up occurs at
```

Comment:
Would use of `defer` in each individual test method or a testing suite not have been sufficient?

Reply:
Well, I only wanted to disconnect the client once... after all the tests completed. Is there a way to do this?

Reply:
You don't have to worry about it now, but a testing suite would've offered this functionality. Also, if you used a mock, this whole process would've been quite a bit more simple because there would've been no penalty for "disconnecting," since that would've been a NOP.